File: lib/platform/android/StatusPresentation.kt
package com.panda.tracker.tracker.widget

import androidx.annotation.ColorRes
import com.panda.tracker.tracker.R

/**
 * The exact color/glyph/label vocabulary from `DESIGN.md`'s color tokens
 * (Story 5.2 Dev Notes table). Every widget cell's color/glyph/label is a
 * direct lookup from this table keyed on the `status` string already
 * present in the decoded JSON — no arithmetic, no comparison against a
 * target, no re-implementation of any evaluation concept (AD-7).
 */
data class StatusPresentation(
    @ColorRes val backgroundColorRes: Int,
    @ColorRes val onColorRes: Int,
    val glyph: String,
    val labelWord: String,
)

private val statusPresentations = mapOf(
    "success" to StatusPresentation(
        R.color.status_success_bg,
        R.color.status_success_on,
        "✓",
        "Success",
    ),
    "fail" to StatusPresentation(
        R.color.status_fail_bg,
        R.color.status_fail_on,
        "✕",
        "Failed",
    ),
    "cheat" to StatusPresentation(
        R.color.status_cheat_bg,
        R.color.status_cheat_on,
        "C",
        "Cheat day used",
    ),
    "pending" to StatusPresentation(
        R.color.status_pending_bg,
        R.color.status_pending_on,
        "…",
        "Pending",
    ),
    "empty" to StatusPresentation(
        R.color.status_empty_bg,
        R.color.status_empty_on,
        "–",
        "Not scheduled",
    ),
)

/** Falls back to the "empty" treatment for any unrecognized status string. */
fun presentationFor(status: String): StatusPresentation =
    statusPresentations[status] ?: statusPresentations.getValue("empty")

/**
 * "<goal>, <state>" for the Today widget's per-goal dot; "<goal>, <date>,
 * <state>" for Week/Month grid cells, matching EXPERIENCE.md's Accessibility
 * Floor pattern so a screen-reader user gets both axes.
 */
fun statusLabel(goalName: String, status: String, date: String? = null): String {
    val word = presentationFor(status).labelWord
    return if (date != null) "$goalName, $date, $word" else "$goalName, $word"
}

File: lib/platform/ios/StatusPresentation.swift
import SwiftUI

/// The exact color/glyph/label vocabulary from `DESIGN.md`'s color tokens
/// (Story 5.2 Dev Notes table). Every widget cell's color/glyph/label is a
/// direct lookup from this table on the `status` string already present in
/// the decoded JSON — no arithmetic, no comparison, no re-implementation of
/// any evaluation concept (AD-7). Colors reference Asset Catalog color sets
/// (Any/Dark appearance variants) so both platforms encode the identical
/// hex values verbatim, light and dark, per Task 4.1's parity guardrail.
struct StatusPresentation {
    let backgroundColorName: String
    let onColorName: String
    let glyph: String
    let labelWord: String
}

enum StatusPresentationCatalog {
    static let presentations: [String: StatusPresentation] = [
        "success": StatusPresentation(
            backgroundColorName: "StatusSuccessBg",
            onColorName: "StatusSuccessOn",
            glyph: "✓",
            labelWord: "Success"
        ),
        "fail": StatusPresentation(
            backgroundColorName: "StatusFailBg",
            onColorName: "StatusFailOn",
            glyph: "✕",
            labelWord: "Failed"
        ),
        "cheat": StatusPresentation(
            backgroundColorName: "StatusCheatBg",
            onColorName: "StatusCheatOn",
            glyph: "C",
            labelWord: "Cheat day used"
        ),
        "pending": StatusPresentation(
            backgroundColorName: "StatusPendingBg",
            onColorName: "StatusPendingOn",
            glyph: "…",
            labelWord: "Pending"
        ),
        "empty": StatusPresentation(
            backgroundColorName: "StatusEmptyBg",
            onColorName: "StatusEmptyOn",
            glyph: "–",
            labelWord: "Not scheduled"
        ),
    ]

    static func presentation(for status: String) -> StatusPresentation {
        presentations[status] ?? presentations["empty"]!
    }

    /// "<goal>, <state>" for the Today widget's per-goal dot; "<goal>,
    /// <date>, <state>" for Week/Month grid cells, matching EXPERIENCE.md's
    /// Accessibility Floor pattern.
    static func label(goalName: String, status: String, date: String? = nil) -> String {
        let word = presentation(for: status).labelWord
        if let date {
            return "\(goalName), \(date), \(word)"
        }
        return "\(goalName), \(word)"
    }
}
